controllers/istiorevision/istiorevision_controller.go

```diff
@@ -372,6 +408,109 @@ func (r *IstioRevisionReconciler) determineReadyCondition(ctx context.Context,
 	}, nil
 }
 
+func (r *IstioRevisionReconciler) determineInUseCondition(ctx context.Context, rev *v1alpha1.IstioRevision) (v1alpha1.IstioRevisionCondition, error) {
+	isReferenced, err := r.isRevisionReferencedByWorkloads(ctx, rev)
+	if err != nil {
+		return v1alpha1.IstioRevisionCondition{}, err
+	}
+
+	if isReferenced {
+		return v1alpha1.IstioRevisionCondition{
+			Type:    v1alpha1.IstioRevisionConditionTypeInUse,
+			Status:  metav1.ConditionTrue,
+			Reason:  v1alpha1.IstioRevisionConditionReasonReferencedByWorkloads,
+			Message: "Referenced by at least one pod or namespace",
+		}, nil
+	}
+	return v1alpha1.IstioRevisionCondition{
+		Type:    v1alpha1.IstioRevisionConditionTypeInUse,
+		Status:  metav1.ConditionFalse,
+		Reason:  v1alpha1.IstioRevisionConditionReasonNotReferenced,
+		Message: "Not referenced by any pod or namespace",
+	}, nil
+}
+
+func (r *IstioRevisionReconciler) isRevisionReferencedByWorkloads(ctx context.Context, rev *v1alpha1.IstioRevision) (bool, error) {
+	logger := log.FromContext(ctx)
+	nsList := corev1.NamespaceList{}
+	nsMap := map[string]corev1.Namespace{}
+	if err := r.Client.List(ctx, &nsList); err != nil { // TODO: can we optimize this by specifying a label selector
+		return false, err
+	}
+	for _, ns := range nsList.Items {
+		if namespaceReferencesRevision(ns, rev) {
+			logger.Info("Revision is referenced by Namespace", "Namespace", ns.Name)
```

Comment:
nit: maybe these log.info are too much verbosity to have by default? maybe turn them debug messages instead?

Reply:
+1 done